src/olympia/promoted/tests/test_commands.py

```diff
@@ -412,3 +412,38 @@ def disable_post_save_promoted_addon(self):
             sender=self.promoted_addon_class,
             dispatch_uid='addons.sync_promoted.promoted_addon_proxy',
         )
+
+
+class TestPromoteByFirefoxThemesCommand(TestCase):
+    def test_basic(self):
+        user = user_factory(pk=settings.TASK_USER_ID)
+        non_affected = [
+            # Not By Firefox user.
+            addon_factory(type=amo.ADDON_STATICTHEME),
+            # Not a theme.
+            addon_factory(users=[user]),
+            # Not public.
+            addon_factory(
+                type=amo.ADDON_STATICTHEME, status=amo.STATUS_DISABLED, users=[user]
+            ),
+        ]
+        # Already promoted, should not cause an error.
+        addon_factory(type=amo.ADDON_STATICTHEME, users=[user], promoted_id=LINE.id)
+
+        expected_affected = addon_factory(type=amo.ADDON_STATICTHEME, users=[user])
+
+        call_command('promote_by_firefox_themes')
+
+        for addon in non_affected:
```

Comment:
We have already migrating reads from `PromotedAddon` to `PromotedAddonPromotion`, so to be consistent, query against `PromotedAddonPromotion` and `PromotedApproval` instead.